pkg/operator/controller/dns/controller.go

```diff
@@ -588,6 +588,28 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 			return nil, fmt.Errorf("failed to create IBM DNS manager: %v", err)
 		}
 		dnsProvider = provider
+	case configv1.PowerVSPlatformType:
```

Comment:
If this is identical to the case for configv1.IBMCloudPlatformType, i.e. you can reuse the `CISCRN: platformStatus.IBMCloud.CISInstanceCRN` please combine them into one case.

Reply:
Since IBMCloud and PowerVS platform are two different platforms, CISCRN for PowerVS platform will available under `platformStatus.PowerVS.CISInstanceCRN` so we cannot combine the switch cases.

Reply:
Created a common function